### ISSUE TITLE
Does the string_list have to be public?

### ISSUE THREAD

**Issue by m-kru** [bpadalino/vhdl-format#4]
Does the `string_list` have to be public? If it is not intended to be used by the user, then it probably should be hidden within package body.

**Comment by bpadalino** [comment]
I exposed it because of [this procedure](https://github.com/bpadalino/vhdl-format/blob/1b3dca65e423e27afef78a8017313a38299161be/fmt.vhd#L103).  This allows any number of string arguments because the line is built outside of the procedure call.

I thought it would be useful in general to have it, since you can also use the string list to build up a bunch of formatted strings, and then concatenate them all together yourself.

What do you think?

**Comment by m-kru** [comment]
Personally I like when the code is compact and orthogonal (no multiple ways for doing the same thing). If one have to format string with more than 16 formatters (which is very very unlikely), one can concatenate calls to `f`. It is probably even more concise than building `string_list` by hand.

**Comment by bpadalino** [comment]
I'll keep the issue open for now, but I see utility in keeping it public and giving options.

There's already multiple ways to perform the same operation:
```
f("{} is a real number", f(3.14159,"10.6f"))
f("{:10.6f} is a real number", 3.14159)
f("{:10.6f} is a real number", f(3.14159))
```

... all produce the same string.  So making the code compact and orthogonal, I think, isn't compelling enough.

**Comment by m-kru** [comment]
These are not the same cases.

1. `f("{} is a real number", f(3.14159,"10.6f"))` is one way because you have to support formatting string within a string. This is simply a side effect.
2. The goal of `f("{:10.6f} is a real number", 3.14159)` is to reduce verbosity and avoid overload conflicts in case of functions with single argument.
3. `f("{:10.6f} is a real number", f(3.14159))` is possible because you have to provide functions for standard type conversion anyway. This is also side effect.

In case of `f` with variable length list its main goal causes the redundancy. In all of the above examples the redundancy is a side effect, that you cannot avoid.

**Comment by bpadalino** [comment]
As of 3a1aca3016fdaba6afb4cfdff33c0c245ce88bda, I decided to keep it public but in a separate package.  I also renamed the package from `fmt` to `fmt_p`, which I personally like better.  I made `fpr` into `fmt` so instead of thinking of the function as "f print" it's more like "format" which seems more concise.

The `string_list_p` contains the `string_list` type and all the functions now, but everything is still contained within the same `fmt.vhd` file - just a separate package.

Going to close down this issue for now.  Feel free to raise more of them, though.
